pkgs/applications/misc/digitalbitbox/default.nix

```diff
@@ -0,0 +1,109 @@
+{
+  stdenv,
+  autoconf,
+  automake,
+  curl,
+  fetchFromGitHub,
+  gcc5,
+  git,
+  libevent,
+  libtool,
+  libqrencode,
+  libudev,
+  libusb,
+  makeWrapper,
+  pkgconfig,
+  qt59,
+  withDebug ? false
+}:
+
+let
+  gcc = gcc5;
+  qt = qt59;
+in stdenv.mkDerivation rec {
+  name = "dbb-app-${version}";
+
+  meta = with stdenv.lib; {
+    description = "A QT based application for the Digital Bitbox hardware wallet";
+    homepage = "https://digitalbitbox.com/";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [
+      vidbina
+    ];
+  };
+
+  src = fetchFromGitHub {
+    owner = "digitalbitbox";
+    repo = "dbb-app";
+    rev = "v${version}";
+    sha256 = "1r77fvqrlaryzij5dfbnigzhvg1d12g96qb2gp8dy3xph1j0k3s1";
+  };
+
+  version = "2.2.2";
+
+  buildInputs = with stdenv.lib; flatten [
+    reqPkgs
+    gccPkgs
+    utilPkgs
+  ];
+
+  reqPkgs = [
+    curl
+    libevent
+    libtool
+    libudev
+    libusb
+    libqrencode
+    pkgconfig
+    qt.full
+  ];
+
+  gccPkgs = [
+    autoconf
+    automake
+    gcc
+  ];
+
+  utilPkgs = [
+    git
+  ];
+
+  QTDIR="${qt.qtbase.dev}";
+
+  MOC="${qt.qtbase.dev}/bin/moc";
+  UIC="${qt.qtbase.dev}/bin/uic";
+  RCC="${qt.qtbase.dev}/bin/rcc";
+  LRELEASE="${qt.qttools.dev}/bin/lrelease";
+  LUPDATE="${qt.qttools.dev}/bin/lupdate";
+
+  configurePhase = with stdenv.lib; let
+    debugFlag = optionalString withDebug "--enable-debug";
+  in ''
+    ./autogen.sh
+    ./configure --prefix=$out ${debugFlag} --enable-libusb $configureFlags
+  '';
+
+  hardeningDisable = [
+    "format"
+  ];
+
+  installPhase = ''
+    make install
```

Comment:
Consider also using the default installPhase (which is `make install`) and add additional stuff with `postInstall`. I'm not exactly sure on this though.